### DISCUSSION TITLE
is there I way to run code after response? having access to req and res context?

### DISCUSSION THREAD

**Discussion by leognmotta** [honojs/hono#3055] [Q&A] [answered]
I need a way to run some code after we have finished some operation and are ready to send response, is it possible?

**Comment by NicoPlyley** [comment, marked as answer]
Hi @leognmotta,
I think what you are looking for the middleware, you will have access to the context as well. In middleware everything after `await next()` will run after the controller returns a responses and before it sent to the client take a look at the docs [here](https://hono.dev/docs/guides/middleware)

**Reply by leognmotta** [reply]
@NicoPlyley i haven't noticed that, it is exactly what I am was looking for, thx.

**Reply by NicoPlyley** [reply]
No problem, will you go ahead and mark the answer to close this out

**Reply by leognmotta** [reply]
> No problem, will you go ahead and mark the answer to close this out

Yes, thanks for helping out
